clinica/pydra/interfaces.py

```diff
@@ -3,9 +3,59 @@
 
 import nipype.interfaces.io as nio
 import pydra
+from nipype.interfaces.base import Directory, DynamicTraitedSpec, Str, isdefined, traits
+from nipype.interfaces.io import IOBase, add_traits
 from pydra.tasks.nipype1.utils import Nipype1Task
 
 
+class CAPSDataGrabberInputSpec(DynamicTraitedSpec):
+    base_dir = Directory(exists=True, desc="Path to CAPS Directory.", mandatory=True)
+    output_query = traits.Dict(
+        key_trait=Str, value_trait=traits.Dict, desc="Queries for outfield outputs"
+    )
+    raise_on_empty = traits.Bool(
+        True,
+        usedefault=True,
+        desc="Generate exception if list is empty for a given field",
+    )
+
+
+class CAPSDataGrabber(IOBase):
+    input_spec = CAPSDataGrabberInputSpec
+    output_spec = DynamicTraitedSpec
+
+    def __init__(self, **kwargs):
+        super(CAPSDataGrabber, self).__init__(**kwargs)
+
+        if not isdefined(self.inputs.output_query):
+            self.inputs.output_query = {}
+
+        # used for mandatory inputs check
+        undefined_traits = {}
+        self.inputs.trait_set(trait_change_notify=False, **undefined_traits)
+
+    def _list_outputs(self):
+        from clinica.utils.inputs import clinica_file_reader
+        from clinica.utils.participant import get_subject_session_list
+
+        sessions, subjects = get_subject_session_list(
+            self.inputs.base_dir,
+            is_bids_dir=False,
+        )
+        return {
+            k: clinica_file_reader(
```

Comment:
There should be a distinction between files read by `clinica_file_reader` and those read by `clinica_group_reader`(templates for example)

Reply:
Yes, I'm working on that atm. Thanks!